### ISSUE TITLE
S3 Requester Pays Does not work with TransferManager

### ISSUE THREAD

**Issue by jooybar** [aws/aws-sdk-java#1175]
Error message:
```
Service: Amazon S3; Status Code: 403; Error Code: 403 Forbidden; Request ID: 67D325E721442841
```

How to reproduce:
1- Create a bucket with all permissions enabled.
2- Enable Requester Pays flag for this bucket
3- Try to download a file using `
Download com.amazonaws.services.s3.transfer.TransferManager.download(GetObjectRequest getObjectRequest, File file)`

Here is a sample code to show the problem:
```
import java.io.*;
import com.amazonaws.auth.*;
import com.amazonaws.services.s3.*;
import com.amazonaws.services.s3.model.*;
import com.amazonaws.services.s3.transfer.*;
import com.amazonaws.util.IOUtils;
public class RequesterPaysTest {
    final static AWSCredentials awsCredentials = new BasicAWSCredentials(MY IAMAccessKey,MY IAMSecretKey);
    final static String bucketName = "Name of requester pays bucket";
    final static String fileName = "Name of a file in this bucket ";
    final static AWSCredentialsProvider awsCredentialsProvider =  new AWSStaticCredentialsProvider(awsCredentials);
    final static AmazonS3 s3client = AmazonS3ClientBuilder.standard().withCredentials(awsCredentialsProvider).withRegion(MY bucketRegion).build();
    final static GetObjectRequest getRequest = new GetObjectRequest(bucketName, fileName, true);

    static void testSimpleRequesterPays() {
        try {
        S3Object object = s3client.getObject(getRequest);
        InputStream objectData = object.getObjectContent();
        FileOutputStream out = new FileOutputStream (new File(fileName));
        IOUtils.copy(objectData, out);
        out.close();
        System.out.println(" Simple RequesterPays successful");
        } catch (Exception e) {
        System.out.println(" Simple RequesterPays unsuccessful: " + e.getMessage());
        }
    }
    static void testTransferManagerRequesterPays() {
        try {
            TransferManager tx = TransferManagerBuilder.standard().withS3Client(s3client).build();
            Download download = tx.download(getRequest, new File(fileName));
            while (download.isDone() == false)
                Thread.sleep(10);
            System.out.println(" TransferManager RequesterPays successful");
        } catch (Exception e) {
        System.out.println(" TransferManager RequesterPays unsuccessful: " + e.getMessage());
        }
    }
public static void main(String[] args) throws IOException {
    testSimpleRequesterPays();
    testTransferManagerRequesterPays();
}
}
```

Output is:
```
 Simple RequesterPays successful
 TransferManager RequesterPays unsuccessful: Forbidden (Service: Amazon S3; Status Code: 403; Error Code: 403 Forbidden; Request ID: 67D325E721442841)
```

**Comment by dagnir** [comment]
Hi @jooybar can you tell us what version of the SDK you're using?

**Comment by jooybar** [comment]
@dagnir 1.11.136
Here is maven dependency:
```
<!-- https://mvnrepository.com/artifact/com.amazonaws/aws-java-sdk-s3 -->
		<dependency>
		    <groupId>com.amazonaws</groupId>
		    <artifactId>aws-java-sdk-s3</artifactId>
		    <version>1.11.136</version>
		</dependency>
```

**Comment by varunnvs92** [comment]
I tried with the version 1.11.136 and could not reproduce the issue. I tried with both serial and parallel download code paths.
What is the size of the object you are downloading?
Can you enable wire logging and share it? (DO NOT include Sensitive data)

**Comment by jooybar** [comment]
I am going to collect a wire log with wireshark today (if this is what you meant).
Meanwhile, want to double check what permission you have used. Here is my bucket policy:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": "*"
            },
            "Action": "*",
            "Resource": "arn:aws:s3:::MY.BUCKET/*"
        },
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": "*"
            },
            "Action": "*",
            "Resource": "MY.BUCKET"
        }
    ]
}
```
Also, for this test, make sure that you are using AWS credentials of someone other than bucket owner (lets say another account wants to access your bucket and pays himself).

**Comment by varunnvs92** [comment]
I am able to reproduce the issue. Working on a fix now. The fix should be available within a day or two. Thanks for reporting the bug.

>> I am going to collect a wire log with wireshark today (if this is what you meant).
I wanted to see the http logs we send/receive to S3. If you are using log4j, we have a guide [here](http://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/java-dg-logging.html). But as I can repo the issue, I don't need the logs anymore.

**Comment by varunnvs92** [comment]
The bug is fixed in our recent release (1.11.140). Please test it and let us know if you still have issues.
